### DISCUSSION TITLE
The installation of dependency imagemagick via apk and php extension imagick

### DISCUSSION THREAD

**Discussion by Apelious** [nextcloud/all-in-one#4053] [Questions] [answered]
### **Is it really necessary to install imagick and the imagemagick every time start the nextcloud-aio-nextcloud container?**

![1](https://github.com/nextcloud/all-in-one/assets/57549957/a7283785-0239-4fc5-a2db-fa6bb076d249)

I noticed that the /start.sh script from nextcloud-aio-nextcloud container, it will read two environment variables from docker run: **$ADDITIONAL_APKS=true** and **$ADDITIONAL_PHP_EXTENSIONS=true**, these two environment variables will trigger imagick and the After the installation of imagemagick is completed, only two flag files will be left in the non-persistent layer of the container: **/additional-apks-are-installed** and **/additional-php-extensions-are-installed**.

Since these two files will be cleared after the container is restarted, **$ADDITIONAL_APKS=true** and **$ADDITIONAL_PHP_EXTENSIONS=true** will still be passed in. Therefore, the installation process is repeated after every container restart.

To make matters worse, my network environment is very poor and I will be stuck here for more than two hours, which is very annoying.
![Screenshot_20240111_224455](https://github.com/nextcloud/all-in-one/assets/57549957/b21e6bc7-5019-4535-aea5-6f69199996af)

### Expected behavior 
imagick and the imagemagick should only be installed once

### Actual behavior 
They are installed repeatedly with every reboot

### Host OS 
debian12 on X86_64

#### Nextcloud AIO version 
27.1.5.1

#### Current channel 
the latest channel since 01/12/2024

**Comment by szaimen** [comment, marked as answer]
> my network environment is very poor 

This is what you should actually try to fix since it will cause all kinds of errors the way along.

However I've created https://github.com/nextcloud/all-in-one/pull/4048 to improve the behaviour.

**Reply by szaimen** [reply]
This is now released with v7.11.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by Apelious** [comment]
> > my network environment is very poor
> 
> This is what you should actually try to fix since it will cause all kinds of errors the way along.
> 
> However I've created #4048 to improve the behaviour.

Thank you for your support on my issue!!!
